extensions/wikia/UserProfilePageV3/UserTagsStrategy.class.php

```diff
@@ -0,0 +1,224 @@
+		if ( in_array( 'staff', $this->usersGlobalGroups ) ) {
+			return false;
+		}
+
+		// check if the user is blocked locally, if not, also check if they're blocked globally (via Phalanx)
+		if ( $this->user->isBlocked( true /* use slave DB */, false /* don't log in PhalanxStats */ ) ||
+			$this->user->isBlockedGlobally()
+		) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Returns if the "FOUNDER" tag should be displayed
+	 * i.e. if the user founded the wiki and is still a bureaucrat or admin
+	 * @return boolean
+	 * @author Andrzej 'nAndy' Łukaszewski
+	 */
+	protected function shouldShowFounderTag() {
+		$wiki = WikiFactory::getWikiByID( $this->wg->CityId );
+		if ( intval( $wiki->city_founding_user ) === $this->user->getId() ) {
+			// mech: BugId 18248
+			return !empty( array_intersect( [ 'bureaucrat', 'sysop' ], $this->usersLocalGroups ) );
+		}
+
+		return false;
+	}
+
+	/**
+	 * Returns if the "BANNED FROM CHAT" tag should be displayed
+	 * i.e. if the user is banned from chat
+	 * @return boolean
+	 */
+	protected function shouldShowBannedFromChatTag() {
+		return $this->wg->EnableChat && ( new ChatUser( $this->user ) )->isBanned();
+	}
+
+	/**
+	 * Tries to get at most $requested user tags for this user's global groups
+	 * @param int $requested Display at most this many global tags
+	 * @return array Text of the tags, safe for HTML output
+	 */
+	protected function getGlobalTags( $requested = 1 ) {
+		$tags = [];
+		// Don't try to show too many tags
+		$count = min( $requested, static::MAX_USER_TAGS, count( $this->globalGroupsWithTags ) );
+		for ( $i = 0; $i < $count; $i++ ) {
+			switch ( array_shift( $this->globalGroupsWithTags ) ) {
+				case 'staff':
+					$tags[] = wfMessage( 'user-identity-box-group-staff' )->escaped();
+					break;
+				case 'helper':
+					$tags[] = wfMessage( 'user-identity-box-group-helper' )->escaped();
+					break;
+				case 'vstf':
+					$tags[] = wfMessage( 'user-identity-box-group-vstf' )->escaped();
+					break;
+				case 'voldev':
+					$tags[] = wfMessage( 'user-identity-box-group-voldev' )->escaped();
+					break;
+				case 'wikiastars':
+					$tags[] = wfMessage( 'user-identity-box-group-wikiastars' )->escaped();
+					break;
+				case 'vanguard':
+					$tags[] = wfMessage( 'user-identity-box-group-vanguard' )->escaped();
+					break;
+				case 'council':
+					$tags[] = wfMessage( 'user-identity-box-group-council' )->escaped();
+					break;
+			}
+		}
+
+		return $tags;
+	}
+
+	/**
+	 * Tries to get at most $requested user tags for this user's local groups
+	 * @param int $requested Display at most this many local tags
+	 * @return string Text of the tag, safe for HTML output
+	 */
+	protected function getLocalTags( $requested = 1 ) {
+		$tags = [];
+		// Don't try to show too many tags
+		$count = min( $requested, static::MAX_USER_TAGS, count( $this->localGroupsWithTags ) );
+		for ( $i = 0; $i < $count; $i++ ) {
+			switch ( array_shift( $this->localGroupsWithTags ) ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a "case default" clause to this "switch" statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)